aea/cli/add.py

```diff
@@ -46,29 +53,49 @@
     PublicId,
     SkillConfig,
 )
-from aea.configurations.constants import CONNECTION, CONTRACT, PROTOCOL, SKILL
+from aea.configurations.constants import AGENT, CONNECTION, CONTRACT, PROTOCOL, SKILL
 from aea.exceptions import enforce
 
 
 @click.group()
 @registry_flag()
+@click.option(
```

Comment:
Should these be merged? I.e. the flag be added to the registry flag? Also, why not simply have `local` and `remote` and then different `remote` providers (ipfs, etc)?

Need to add a type for Union[PublicId, IPFSHash]